src/main/java/de/tum/cit/aet/artemis/programming/service/ProgrammingExerciseParticipationService.java

```diff
@@ -181,11 +213,21 @@ public ProgrammingExerciseStudentParticipation findStudentParticipationByExercis
 
         Optional<ProgrammingExerciseStudentParticipation> participationOptional;
 
-        if (withSubmissions) {
-            participationOptional = studentParticipationRepository.findWithSubmissionsByExerciseIdAndStudentLoginAndTestRun(exercise.getId(), username, isTestRun);
+        if (exercise.isTestExamExercise()) {
```

Comment:
Why can't we always use the version with findFirst? If there is only one real exam, it should not make a difference.

Reply:
good point, thank you. You're right, we can use `findFirst`, it doesn't make any difference.